kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -534,6 +535,24 @@ def _report_ephemeral_storage_usage(self, pod_list, instance_tags):
 
             self.gauge(self.NAMESPACE + '.ephemeral_storage.usage', pod_usage, tags)
 
+    def _report_system_container_metrics(self, stats, instance_tags):
+        sys_containers = stats.get('node', {}).get('systemContainers', [])
+        for ctr in sys_containers:
+            if ctr.get('name') == 'runtime':
+                mem_rss = ctr.get('memory', {}).get('rssBytes')
+                if mem_rss:
+                    self.gauge(self.NAMESPACE + '.runtime.memory.rss', mem_rss, instance_tags)
```

Comment:
Just a note that the base class offers this feature out of the box: https://github.com/DataDog/integrations-core/blob/e98ce28ad47f13a78354d39ec82fde1b9877eaf7/datadog_checks_base/datadog_checks/base/checks/base.py#L595

You just have to set `self.__NAMESPACE__ = 'kubelet'` and all metrics will be prefixed.

Reply:
tests fail if I remove the `self.NAMESPACE` prefix when submitting the metric `self.gauge('.runtime.memory.rss', mem_rss, instance_tags)`

```
>       assert condition, new_msg
E       AssertionError: Needed at least 1 candidates for 'kubernetes.kubelet.cpu.usage', got 0
E         Expected:
E                 MetricStub(name='kubernetes.kubelet.cpu.usage', type=None, value=36755862.0, tags=['instance:tag'], hostname=None, device=None)
E         Similar submitted:
E         Score   Most similar
E         0.87    MetricStub(name='.kubelet.cpu.usage', type=0, value=36755862.0, tags=['instance:tag'], hostname='', device=None)
```

I'm I missing something?

Reply:
You have to set `self.__NAMESPACE__ = 'kubelet'` in the __init__ method and submit the metric without the leading `.` and that should "just work"

Reply:
Thanks for the trick, but this requires making significant changes in the check.

Currently all metrics are submitted with appending the prefix and most of the tests are asserting the `gauge` / `rate` calls and arguments instead of using `aggregator.assert_metric`.
Adding `self.__NAMESPACE__` requires changing all the metric submitting calls in the check and the unit tests as well.

I believe it's not in the scope of this PR to make all these changes but I'm adding a card to make them in a dedicated PR.

Reply:
Sounds good, I was just pointing to some convenience method here 👍